### PR TITLE
GH-174: Synchronizing between current cursor in editor and node in outline tree view is broken

### PR DESCRIPTION
**Pull request by qtran-n4**
## Task
https://github.com/eclipse/n4js/issues/174

In addition to fixing the link with editor problem, this PR also activates the filter actions for JSX language.

### CODE REVIEW

**Comment by qtran-n4** [comment]
@jpilgrim Please review the PR. Thanks! 
BTW,  I wanted to create an SWTBot test that moves the cursor to a say method body and verifies that the selected item in the outline tree is the method and not its parent but unfortunately I haven't found a way to do that in SWTBot test.

**Comment by qtran-n4** [comment]
@yoosiba Please pay attention to the part where `astElement` is called. The important thing here is that we do NOT want to call `astElement` when the AST is potentially is a proxy.

File: plugins/org.eclipse.n4js.n4jsx.ui/src/org/eclipse/n4js/n4jsx/ui/N4JSXUiModule.xtend [hunk 1]
```diff
@@ -561,11 +566,74 @@ class N4JSXUiModule extends AbstractN4JSXUiModule {
 	def Class<? extends IReferenceFilter> bindContentReferenceFilter() {
 		return N4JSXReferencesFilter;
 	}
-	
+
+	/**
+	 * Bind the {@link IReferenceFinder} that find references solely to types (and
+-    * TVariables, IdentifiableElement and TEnumLiterals).
+-    */
```
Comment (yoosiba) [inline plugins/org.eclipse.n4js.n4jsx.ui/src/org/eclipse/n4js/n4jsx/ui/N4JSXUiModule.xtend]: fix formatting

Comment (qtran-n4) [reply plugins/org.eclipse.n4js.n4jsx.ui/src/org/eclipse/n4js/n4jsx/ui/N4JSXUiModule.xtend]: done!

**Comment by qtran-n4** [comment]
Builds are green. Merging now.

**Pull request merged**